services/state/blockstream/block_stream_info.proto

```diff
@@ -83,6 +86,49 @@ message BlockStreamInfo {
      * hash SHALL be for block number N-256.<br/>
      * The latest available hash SHALL be for block N-1.<br/>
      * This is REQUIRED to implement the EVM `BLOCKHASH` opcode.
+     * <p>
+     * <div style="display: block;font-size: .83em;margin-top: 1.67em;margin-bottom: 1.67em;margin-left: 0;margin-right: 0;font-weight: bold;">Field Length</div>
+     * Each hash value SHALL be the trailing 265 bits of a SHA2-384 hash.<br/>
+     * The length of this field SHALL be an integer multiple of 32 bytes.<br/>
+     * This field SHALL be at least 32 bytes.<br/>
+     * The maximum length of this field SHALL be 8192 bytes.
      */
     bytes trailing_block_hashes = 4;
+
+    /**
+     * A SHA2-384 hash value.<br/>
+     * This is the hash of the "input" subtree for this block.
+     */
+    bytes input_tree_root_hash = 5;
+
+    /**
+     * A SHA2-384 hash value.<br/>
+     * This is the hash of consensus state at the _start_ of this block.
+     */
+    bytes start_of_block_state_hash = 6;
+
+    /**
+     * A count of "output" block items in this block.
+     * <p>
+     * This SHALL count the number of output block items that _precede_
+     * the state change that updates this singleton.
+     */
+    uint32 num_preceding_output_items = 7;
+
+    /**
+     * A concatenation of SHA2-384 hash values.<br/>
+     * This is the "rightmost" values of the "output" subtree.
+     * <p>
+     * The subtree containing these hashes SHALL be constructed from all "output"
+     * `BlockItem`s in this block that _precede_ the update to this singleton.
+     */
+    repeated bytes rightmost_preceding_output_tree_hashes = 8;
```

Comment:
The "output" tree is strictly binary, I would expect that this only ever needs to be a single hash, but perhaps I'm missing something in what this is supposed to be.

Also, wouldn't "preceding" items be "left" of the current item?

Reply:
These are all the right hashes of the binary tree needed to compute indirect block proof.